static/main.ts

```diff
@@ -0,0 +1,111 @@
+import { createAppKit } from "@reown/appkit";
+import { Ethers5Adapter } from "@reown/appkit-adapter-ethers5";
+import { gnosis, mainnet, polygon, optimism, arbitrum, base, bsc, blast, zksync, avalanche, worldchain } from "@reown/appkit/networks";
+import { ethers } from "ethers";
+import { renderErrorInModal } from "./display-popup-modal";
+import { updateTokens } from "./populate-dropdown";
+import { setupApproveButton, setupRevokeButton, setupValidityListener } from "./handle-approval";
+
+// passed in at runtime
+declare const ALCHEMY_KEY: string;
+
+// All unhandled errors are caught and displayed in a modal
+window.addEventListener("error", (event: ErrorEvent) => renderErrorInModal(event.error));
+
+// All unhandled promise rejections are caught and displayed in a modal
+window.addEventListener("unhandledrejection", (event: PromiseRejectionEvent) => {
+  renderErrorInModal(event.reason as Error);
+  event.preventDefault();
+});
+
+const projectId = "415760038f8e330de4868120be3205b8";
+
+const metadata = {
+  name: "Ubiquity Allowance",
+  description: "Allow funding in Ubiquity",
+  url: "https://onboard.ubq.fi",
+  icons: ["https://avatars.githubusercontent.com/u/76412717"],
+};
+
+const alchemyKey = ALCHEMY_KEY || "";
+
+const providersUrl: { [key: string]: string } = {
+  100: alchemyKey ? `https://gnosis-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://rpc.gnosischain.com",
```

Comment:
> Let's remove alchemyKey. It is getting into many files and isn't necessary either. In fact, you probably don't need this or hardcoded rpc because  @reown/appkit/networks has it. 
> 
> ```
> import { gnosis } from "@reown/appkit/networks";
> 
> gnosis.rpcUrls.default.http;
> gnosis.rpcUrls.default.webSocket;
> ```

reown's RPC feature is broken, so cant use it

Reply:
Wouldn't it be better to use user's RPC from the wallet?

Reply:
not every wallet has an rpc, unfortunately